packages/api/endpoints/providers.js

```diff
@@ -131,32 +132,40 @@ async function post(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function put({ params: { id }, body }, res) {
-  if (id !== body.id) {
+  const apiProvider = body;
+
+  if (id !== apiProvider.id) {
     return res.boom.badRequest(
       `Expected provider ID to be '${id}', but found '${body.id}' in payload`
     );
   }
 
-  const knex = await getKnexClient({ env: process.env });
+  const knex = await getKnexClient();
   const providerModel = new Provider();
-
-  const providerExists = await Promise.all([
-    providerModel.exists(id),
-    doesRecordExist({ name: id }, knex, tableNames.providers),
-  ]);
-
-  if (providerExists.filter((providerTest) => providerTest === true).length !== 2) {
+  let oldProvider;
+  try {
+    oldProvider = await providerModel.get({ id });
+  } catch (error) {
+    if (error.name !== 'RecordDoesNotExist') {
+      throw error;
+    }
     return res.boom.notFound(
-      `Provider with ID '${id}' not found in Dynamo and Postgres databases`
+      `Provider with ID '${id}' not found`
     );
   }
 
+  apiProvider.updatedAt = Date.now();
+  apiProvider.createdAt = oldProvider.createdAt;
+
   let record;
-  let createObject = await translateApiProviderToPostgresProvider(body);
-  createObject = nullifyUndefinedProviderValues(createObject);
+  const postgresProvider = await translateApiProviderToPostgresProvider(apiProvider);
+
   await knex.transaction(async (trx) => {
-    await trx(tableNames.providers).where({ name: id }).update(createObject);
-    record = await providerModel.create(body);
+    await trx(tableNames.providers)
```

Comment:
Update behavior here isn't consistent with the dynamodoc client create method, and nullifying values isn't needed as they'll be *explicitly* set as undefined in the translation method, and the knex insert/merge will set them to null as a result.

Reply:
Definitely agree with making this an upsert, but to be consistent we should define an `upsert` method on the provider PG model and use it here

Reply:
41824be97